jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -149,9 +149,8 @@ public void onHandshakeResponse(HttpRequest request, HttpResponse response)
                 }
             });
         }
-        upgradeRequest.setConfiguration(configurationCustomizer);
-        CompletableFuture<Session> futureSession = new CompletableFuture<>();
 
+        CompletableFuture<Session> futureSession = new CompletableFuture<>();
```

Comment:
You don't need to allocate this `CF`, just return what `whenComplete()` returns below.

Reply:
That doesn't seem right.  
We want to return immediately, and let the whenComplete handle the success/exception at a later point in time.

Reply:
This `CF` is giving a Jetty WS API `Session`, and the one with the `whenComplete()` returns a `CoreSession`. They're not interchangeable.
The core connect is used to complete the future for the jetty ws api connect.

Reply:
@lachlan-roberts CFs can be `map()`ed to transform their result; perhaps use `CF.handle()` rather than `whenComplete()`.

Reply:
Just looked at that, but I don't think we can map the exception like we are currently doing using `CF.handle()`.

Reply:
@lachlan-roberts good point about the transformation of the exception.
Code is good as is, returning the CF created on the fly, rather than the one returned by `whenComplete()`.